File: mtg-card-organizer/src/app/card/models/card-instance.ts
import { CardSet } from './card-set';

export class CardInstance {
  id: number;

  foil = false;
  promo = false;

  cardSetId: number;
  cardSet: CardSet;

  imageUrl: string;

  public constructor(init?: Partial<CardInstance>) {
    Object.assign(this, init);
  }
}

File: MtgCardOrganizer.Dal/Entities/Collections/CardInstance.cs
using System.Collections.Generic;
using MtgCardOrganizer.Dal.Entities.Cards;
using MtgCardOrganizer.Dal.Entities.Collections;

namespace MtgCardOrganizer.Dal.Entities.Collections
{
    public class CardInstance : Entity
    {
        public bool Foil { get; set; }
        public bool Promo { get; set; }

        public int CardSetId { get; set; }
        public int CollectionId { get; set; }
        public CardSet CardSet { get; set; }
        public Collection Collection { get; set; }
    }
}
